template/template.go

```diff
@@ -141,6 +102,79 @@ type Pair struct {
 	Name, Value string
 }
 
+// Pairs is a list of key/value string pairs.
+type Pairs []Pair
+
+// Names returns a list of names of the pairs.
+func (ps Pairs) Names() []string {
+	ns := make([]string, 0, len(ps))
+	for _, p := range ps {
+		ns = append(ns, p.Name)
+	}
+	return ns
+}
+
+// Values returns a list of values of the pairs.
+func (ps Pairs) Values() []string {
+	vs := make([]string, 0, len(ps))
+	for _, p := range ps {
+		vs = append(vs, p.Value)
+	}
+	return vs
+}
+
+// KV is a set of key/value string pairs.
+type KV map[string]string
+
+// SortedPairs returns a sorted list of key/value pairs.
+func (kv KV) SortedPairs() Pairs {
+	var (
+		pairs     = make([]Pair, 0, len(kv))
+		keys      = make([]string, 0, len(kv))
+		sortStart = 0
+	)
+	for k := range kv {
+		if k == string(model.AlertNameLabel) {
+			keys = append([]string{k}, keys...)
+			sortStart = 1
+		} else {
+			keys = append(keys, k)
+		}
+	}
+	sort.Strings(keys[sortStart:])
+
+	for _, k := range keys {
+		pairs = append(pairs, Pair{k, kv[k]})
+	}
+	return pairs
+}
+
+// Without returns a copy of the key/value set without the given keys.
```

Comment:
I'm not sure about this function. It isn't related to the `with` statement that's in the templates, and to someone not familiar with Go's templates this looks like a field access rather than a method call.

I think we might be better off keeping this as a separate function.